clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6387,6 +6395,78 @@ static void handleIntelFPGAMaxReplicatesAttr(Sema &S, Decl *D,
   S.AddIntelFPGAMaxReplicatesAttr(D, A, A.getArgAsExpr(0));
 }
 
+// Handle [[intel:fpga_pipeline]] attribute.
+void Sema::AddSYCLIntelFpgaPipelineAttr(Decl *D, const AttributeCommonInfo &CI,
+                                        Expr *E) {
+  if (!E->isValueDependent()) {
+    // Validate that we have an integer constant expression and then store the
+    // converted constant expression into the semantic attribute so that we
+    // don't have to evaluate it again later.
+    llvm::APSInt ArgVal;
+    ExprResult Res = VerifyIntegerConstantExpression(E, &ArgVal);
+    if (Res.isInvalid())
+      return;
+    E = Res.get();
+
+    // This attribute requires a non-negative value.
+    if (ArgVal < 0) {
+      Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+          << CI << /*non-negative*/ 1;
+      return;
+    }
```

Comment:
This doesn't seem right?

Reply:
I have removed this check. The attribute takes only 0 or 1. 
Negative numbers should not be a problem here.  We do not need to check here.
Any nonzero number should be treated as true, and zero should be treated as false. @AaronBallman, Ay thought?

Reply:
Agreed!